### ISSUE TITLE
Space after end do

### ISSUE THREAD

**Issue by hiker** [stfc/PSyclone#611]
Atm psyclone creates 'end do' statements as 'END DO ' - notice the space at the end. While mostly this does not matter, it triggers pylint warnings if a ''' multi-line string is used to compare the output with, e.g. in gocean1p0_config_test:
```
new_loop1 = '''      DO j=1,2
        DO i=3,4
          CALL compute_kern1_code(i, j, cu_fld%data, p_fld%data, u_fld%data)
        END DO 
      END DO '''   # nopep8
```
Since multi-line strings are very convenient for storing Fortran strings, I would prefer if the space at the end of 'end do' could be removed, so that we don't need to add pylint/pycodestyle directives.

The fix must be done in fparser, see https://github.com/stfc/fparser/issues/228 for details. If this is fixed, we need to update the test cases to remove the spaces, which is tracked in this ticket.

**Comment by arporter** [comment]
https://github.com/stfc/fparser/issues/228 is now done.

**Comment by hiker** [comment]
Thanks a lot, @arporter . How will this get into psyclone? Do we need a new release of fparser for that?

**Comment by arporter** [comment]
No worries. We just need to update the fparser submodule within PSyclone. (cd PSyclone/external/fparser; checkout latest master; cd ..; git commit fparser) I saw you'd already created an Issue for this (since it will break lots of tests).

**Comment by hiker** [comment]
Sorry, that was badly formulated - I meant, what is the policy - do we only point to released fparser versions, or not? And you answered that question :) I'm happy to handle that ... shortly.

**Comment by arporter** [comment]
Since I'm updating to the latest fparser in #634 I'll also fix this issue there.

**Comment by hiker** [comment]
*grin* I started on this, but ran into all kind of problems caused by #634 - so I was about to ask for help ... but now I'll just wait patiently ;)  Thanks!

**Comment by hiker** [comment]
This has been merged in recently (#635), closing this issue. Thanks to @arporter for doing this!
